src/app/content/components/PageWithMockedHighlighter.spec.tsx

```diff
@@ -0,0 +1,255 @@
+import mockArchiveLoader, { book, page } from '../../../test/mocks/archiveLoader';
+import createMockHighlight from '../../../test/mocks/highlight';
+import { mockCmsBook } from '../../../test/mocks/osWebLoader';
+import { renderToDom } from '../../../test/reactutils';
+import { resetModules } from '../../../test/utils';
+import SkipToContentWrapper from '../../components/SkipToContentWrapper';
+import * as Services from '../../context/Services';
+import MessageProvider from '../../MessageProvider';
+import { locationChange } from '../../navigation/actions';
+import { scrollTarget } from '../../navigation/selectors';
+import { AppServices, MiddlewareAPI, Store } from '../../types';
+import { assertDocument, assertWindow } from '../../utils';
+import { receivePage } from '../actions';
+import {
+  clearFocusedHighlight,
+  focusHighlight,
+  receiveDeleteHighlight,
+  receiveHighlights,
+  requestDeleteHighlight,
+} from '../highlights/actions';
+import * as requestRemoveHighlight from '../highlights/hooks/requestRemoveHighlight';
+import { initialState } from '../reducer';
+import { formatBookData } from '../utils';
+import ConnectedPage from './Page';
+import allImagesLoaded from './utils/allImagesLoaded';
+
+jest.mock('./utils/allImagesLoaded', () => jest.fn());
+jest.mock('../highlights/components/utils/showConfirmation', () => () => new Promise((resolve) => resolve(false)));
+
+// https://github.com/facebook/jest/issues/936#issuecomment-463644784
+jest.mock('../../domUtils', () => ({
+  // remove cast to any when the jest type is updated to include requireActual()
+  ...(jest as any).requireActual('../../domUtils'),
+  scrollTo: jest.fn(),
+}));
+
+jest.mock('@openstax/highlighter');
+
+UntypedHighlighter.prototype.eraseAll = jest.fn();
+UntypedHighlighter.prototype.erase = jest.fn();
+UntypedHighlighter.prototype.highlight = jest.fn();
+
+// tslint:disable-next-line:variable-name
+const Highlighter = UntypedHighlighter as unknown as jest.SpyInstance;
+const fromApiResponse = UntypedSerializedHighlight.fromApiResponse = jest.fn();
+
+describe('Page', () => {
+  let archiveLoader: ReturnType<typeof mockArchiveLoader>;
+  let store: Store;
+  let dispatch: jest.SpyInstance;
+  let services: AppServices & MiddlewareAPI;
+
+  beforeEach(() => {
+    resetModules();
+    jest.resetAllMocks();
+
+    // scrollIntoView is not implemented in jsdom
+    (assertWindow() as any).HTMLElement.prototype.scrollIntoView = () => jest.fn();
+
+    (allImagesLoaded as any as jest.SpyInstance).mockReturnValue(Promise.resolve());
+
+    store = createTestStore({
+      content: {
+        ...initialState,
+        book: formatBookData(book, mockCmsBook),
+        page,
+      },
+    });
+    dispatch = jest.spyOn(store, 'dispatch');
+
+    const testServices = createTestServices();
+
+    services = {
+      ...testServices,
+      dispatch: store.dispatch,
+      getState: store.getState,
+    };
+    archiveLoader = testServices.archiveLoader;
```

Comment:
Is it worth having it as a separate variable if it's used in only one test?

Reply:
Probably doesn't matter
It is all about types - I can't use `services.archiveLoader.mockPage` because it does not exist on type `(AppServices & MiddlewareAPI)['archiveLoader']` so I would have to cast it
I've just copied this part from `Page.spec.tsx`